### PR TITLE
Support Swift Package Manager

### LINKED ISSUE #4: Support Swift Package Manager
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/LaunchAtLogin/issues/4)

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $80.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($80.00)
### Submitted pull Requests
- [#25 Support Swift Package Manager](https://issuehunt.io/r/sindresorhus/LaunchAtLogin/pull/25)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/LaunchAtLogin/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by benkoska**
Fixes #4

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#4: Support Swift Package Manager](https://issuehunt.io/repos/98812730/issues/4)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

**Comment by sindresorhus** [comment]
You also need to update the docs.

**Comment by DeveloperBen** [comment]
Ok.

**Comment by DeveloperBen** [comment]
You can't access the file copy-helper.sh file from the repository because it's in .build/checkouts/LaunchAtLogin.git--XXXXXXXX/LaunchAtLogin/copy-helper.sh and you don't know what the XXXXXXXX is.

**Comment by sindresorhus** [comment]
That’s why the issue has a bounty. Adding the Package.swift file is the easy part.

**Comment by sindresorhus** [comment]
Are you sure it's like that with Xcode 11? I'm asking, as I found this: https://github.com/apple/swift-package-manager/commit/ecb971676811f8415a895a3715524a118e6e6c71 Alternatively, there might be an environment variable that provides the location.

**Comment by DeveloperBen** [comment]
I'll download Xcode 11 and try it.

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import PackageDescription
+
+let package = Package(name: "LaunchAtLogin",
+                      products: [.library(name: "LaunchAtLogin",
+                                          targets: ["LaunchAtLogin", "LaunchAtLoginHelper"])],
+                      targets: [.target(name: "LaunchAtLogin",
+                                        path: "LaunchAtLogin"),
+                                        .target(name: "LaunchAtLoginHelper",
+                                        path: "LaunchAtLoginHelper")])
```
Comment (sindresorhus) [inline Package.swift]: This is messy. Try to format it better and use tab-indentation.

Comment (DeveloperBen) [reply Package.swift]: Ok

**Comment by DeveloperBen** [comment]
When using the Xcode Swift Package Manager it saves it in the Derived Data folder.

**Comment by DeveloperBen** [comment]
And when you just add it to the Package.swift (not Xcode 11) then it's saved into the .build folder.

**Comment by DeveloperBen** [comment]
I just tried it with SwiftPM and it runs the script and I can use the library in my code.

**Comment by sindresorhus** [comment]
Can you create a new Xcode project, add the SwiftPM dependency, and test that it find the copy-helper, correctly copies the helper app to the correct location, and that it actually correctly launches the app at login? Because that's where it failed with CocoaPods, it silently did not launch at login.

**Comment by DeveloperBen** [comment]
Ok, I will try that

**Comment by DeveloperBen** [comment]
It's looking for the app under Resources/LaunchAtLoginHelper.app but for me it's located in Library/LoginItems/LaunchAtLoginHelper.app

**Comment by DeveloperBen** [comment]
image

File: readme.md [hunk 2]
```diff
@@ -27,10 +33,19 @@ github "sindresorhus/LaunchAtLogin"
 
 Add a new ["Run Script Phase"](http://stackoverflow.com/a/39633955/64949) below "Embed Frameworks" in "Build Phases" with the following:
 
+SwiftPM:
+
+```
+${BUILD_ROOT}/../../SourcePackages/checkouts/LaunchAtLogin/LaunchAtLogin/copy-helper.sh
```
Comment (sindresorhus) [inline readme.md]: I would try to find an environment variable where you don't have to use `../..`. That's very fragile to future changes.

https://gist.github.com/gdavis/6670468

Comment (DeveloperBen) [reply readme.md]: I looked for one but I find an updated version of all environment variables.

Comment (sindresorhus) [reply readme.md]: You can just print the environment variables yourself. https://www.google.com/search?q=xcode+print+environment+variables&oq=xcode+print+environ&aqs=chrome.0.0j69i57.3568j0j0&sourceid=chrome&ie=UTF-8

**Comment by sindresorhus** [comment]
> It's looking for the app under Resources/LaunchAtLoginHelper.app

What is looking for it there? This is the location before it's copied I think.

> but for me it's located in Library/LoginItems/LaunchAtLoginHelper.app

That's the correct location.

**Comment by DeveloperBen** [comment]
I just looked through the environment variables and I can't find one that fits our use case.

File: readme.md (hunk 2 above)
Comment (sindresorhus) [reply readme.md]: This path is also not correct. The `BUILD_ROOT` during archiving is not there. It needs 5 levels of `../`. And even then, the script itself needs to be updated to support the SwiftPM locations.

**Comment by sindresorhus** [comment]
Closing for lack of activity.

**Pull request closed without merging**